module3/lessons/Week2/RestfulRoutes.md

```diff
@@ -0,0 +1,174 @@
+{
+    var movies = _context.Movies;
+    return View(movies);
+}
+```
+
+This is the existing method in our controller from the MVC lesson. As we witnessed previously, this method returns a View that displays all the movies in our database.
+
+### Show(int id)
+
+The Show() method takes an integer and returns the Movie record with that ID number from the database. 
+To accomplish this, we need to do the following:
+* Create a local variable that stores the movie from the `context` object.
+* Return a View to display the details of that movie.
+
+Add this code to `MoviesController.cs` file, preferably after the `Index()` method:
+
+```c#
+// GET: /Movies/<id>
+public IActionResult Show(int id)
+{
+    var movie = _context.Movies.Find(id);
+    return View(movie);
+}
+```
+
+### Show View
+
+We have a View for our Index route that displays all the movies in our database: `Index.cshtml`. 
+We will create a View for our Show route that displays a single movie.
+
+From Solution Explorer:
+* Expand the Views folder
+* Right-click the Movies folder
+* Add > New File
+* Name the new file `Show.cshtml`
+
+Add the following code to the `Show.cshtml` file:
+
+```html
+@model Movie
+
+<h1>Movie Details</h1>
+
+<p>Title: @Model.Title</p>
+
+<p>Genre: @Model.Genre</p>
+
+```
+
+We have just created a View for our Show route. However...
+
+### RESTful Annotation
+
+We are ALMOST done. Here's our problem: if we visit `/movies/1` to see the movie with the ID `1`, we get an error. To see the `1` movie, we would have to go to `/movies/show/1`. But this does not match the RESTful pattern from our table above. Instead, we need to add an annotation to make our route RESTful.
+
+Add `[Route("Movies/{id:int}")]` above the Show() method declaration. The final method should look like this:
+
+```c#
+// GET: /Movies/<id>
+[Route("Movies/{id:int}")]
+public IActionResult Show(int id)
+{
+    var movie = _context.Movies.Find(id);
+    return View(movie);
+}
+```
+
+Our Show view should now display the `1` movie correctly.
+
+## Later Lessons
+
+In later lessons, we will continue to build out the remaining routes from the RESTful table.
+
+* **New**: This method offers a form for the user to create a new movie object.
+* **Create**: This method creates a new movie object to add to our database, often in conjunction with the `New()` method.
+* **Edit**: This method offers a form for the user to edit an existing movie object.
+* **Update**: This method updates an existing movie object, often in conjunction with the `Edit()` method.
+* **Destroy**: This method deletes an existing movie object from our database.
+
+## Check for Understanding
+
+1. Which HTTP method(s) map to the following CRUD functions?
+	* Create
+	* Read
+	* Update
+	* Delete
+1. If you had an application for `vehicles`, what are the URI's for the 7 RESTful routes that need to be created?
+1. What is the annotation you will need to add for the `Show()` method in the `vehicles` application?
+
```

Comment:
Random idea you can feel free to take or leave, might be interesting to show some examples of non-restful routes and restful routes and have students pick which are which.